model/model.go

```diff
@@ -63,17 +63,32 @@ func (m Model) Perform() (err error) {
 		return
 	}
 
-	archivePath, err = encryptor.Run(archivePath, m.Config)
+	logger.Infof("Cleanup WorkDir: %s/", m.Config.DumpPath)
+	if err := os.RemoveAll(m.Config.DumpPath); err != nil {
+		logger.Errorf("Cleanup temp dir %s error: %v", m.Config.DumpPath, err)
+	}
+
+	encryptedPath, err := encryptor.Run(archivePath, m.Config)
 	if err != nil {
 		return
 	}
+	if encryptedPath != archivePath {
```

Comment:
This logic is hard to read, I don't like it. We need to find other way to remove the last tmp file.